dre_probe_matcher.py

```diff
@@ -0,0 +1,649 @@
+                                            match_data,
+                                            self.soartech_sid,
+                                            self.soartech_yaml['id'])
+            except:
+                pass
+        match_data = {'alignment': st_align, 'data': match_data}
+        if SEND_TO_MONGO:
+            mid = self.participantId + '_st_' + self.environment.split('.yaml')[0]
+            try:
+                mongo_collection_matches.insert_one({'scenario_id': self.soartech_yaml['id'], 'timestamp': self.timestamp, 'evalNumber': EVAL_NUM, 'evalName': EVAL_NAME, 'data': match_data, 'ta1': 'st', 'env': self.environment.split('.yaml')[0], 'pid': self.participantId, '_id': mid})
+            except:
+                mongo_collection_matches.update_one({'_id': mid}, {'$set': {'scenario_id': self.soartech_yaml['id'], 'timestamp': self.timestamp, 'evalNumber': EVAL_NUM, 'evalName': EVAL_NAME, 'data': match_data, 'ta1': 'st', 'env': self.environment.split('.yaml')[0], 'pid': self.participantId, '_id': mid}})
+        json.dump(match_data, self.output_soartech, indent=4)  
+        # QOL:
+        # go to: O vs. U (intent to treat)
+        # go to: O vs. N (treat)
+        # go to: U vs. W (treat)
+        # then drag U vs. W? (move to safety)
+        # go to: G vs. Y (intent to treat)
+        # go to: X vs. Y (treat)
+        # go to: G vs. H; (treat)
+        # then drag G vs. H? (move to safety)
+        # Z vs. L? (intent to treat)
+        # Z vs. A? (treat)
+        # L vs. M; (treat)
+        # pain meds L vs. M? (supply limit)
+        # VOL:
+        # go to: Y vs. G (intent to treat)
+        # go to: Y vs. X (treat)
+        # go to: G vs. H (treat)
+        # then drag G vs. H? (move to safety)
+        # go to: C vs. P (intent to treat)
+        # go to: C vs. B (treat)
+        # go to: P vs. V; (treat)
+        # then drag P vs. V? (move to safety)
+        # O vs. U? (intent to treat)
+        # O vs. N? (treat)
+        # U vs. W; (treat)
+        # then drag U vs. W? (move to safety)
+                
+
+    def get_scene_by_id(self, scenes, scene_id):
+        for scene in scenes:
+            if scene['id'] == scene_id:
+                return scene
+        return None
+
+
+    def get_next_scene_by_index(self, scenes, this_scene):
+        for i in range(len(scenes)-1):
+            if scenes[i] == this_scene:
+                return scenes[i+1]
+        return None
+
+
+    def match_adept_probes(self):
+        adept_scenes = self.adept_yaml['scenes']
+        last_action_ind_used = 0
+        match_data = []
+        found = 0
+        total = 0
+        looking_for_justify = False
+        repeating_gauze = False
+        stored_next_id = None
+        # Adept has branching, which makes things a little more difficult
+        first_scene_id = self.adept_yaml['first_scene']
```

Comment:
Even though ADEPT always uses `first_scene` in DRE, it is not required in the YAML.  It defaults to the first scene in the YAML.

Reply:
You're right. This now will get the first scene, but will default to the first scene in the scenes list if first_scene is not defined